src/main/main.ts

```diff
@@ -188,214 +189,195 @@ function initializeApp() {
-
-      const appDirName = app.getAppPath();
-      // FIXME: "config.json" 이거 하드코딩하지 말아야 함
-      const configFileName = "config.json";
-
-      // 압축 해제하기 전에 기존 설정 꿍쳐둔다. 나중에 기존 설정 내용이랑 새 디폴트 값들이랑 합쳐야 함.
-      const configPath = path.join(appDirName, configFileName);
-      const bakConfig = JSON.parse(
-        await fs.promises.readFile(configPath, { encoding: "utf-8" })
-      );
-      console.log("The existing configuration:", bakConfig);
-
-      if (process.platform == "win32") {
-        // 윈도는 프로세스 떠 있는 실행 파일을 덮어씌우거나 지우지 못하므로 이름을 바꿔둬야 함.
-        const src = app.getPath("exe");
-        const basename = path.basename(src);
-        const dirname = path.dirname(src);
-        const dst = path.join(dirname, "bak_" + basename);
-        await fs.promises.rename(src, dst);
-        console.log("The executing file has renamed from", src, "to", dst);
-
-        // TODO: temp directory 앞에 9c-updater- 접두어
-        const tempDir = await tmpName();
-
-        // ZIP 압축 해제
-        console.log("Start to extract the zip archive", dlPath, "to", tempDir);
-
-        await extractZip(dlPath, {
-          dir: tempDir,
-          onEntry: (_, zipfile) => {
-            const progress = zipfile.entriesRead / zipfile.entryCount;
-            win?.webContents.send("update extract progress", progress);
-          },
-        });
-        win?.webContents.send("update extract complete");
-        console.log("The zip archive", dlPath, "has extracted to", tempDir);
-        win?.webContents.send("update copying progress");
-        await utils.copyDir(tempDir, extractPath);
-        console.log("Copied extracted files from", tempDir, "to", extractPath);
-        try {
-          await fs.promises.rmdir(tempDir, { recursive: true });
-          console.log("Removed all temporary files from", tempDir);
-        } catch (e) {
-          console.warn(
-            "Failed to remove temporary files from",
-            tempDir,
-            "\n",
-            e
-          );
-        }
-        win?.webContents.send("update copying complete");
-      } else if (process.platform == "darwin") {
-        // .tar.{gz,bz2} 해제
-        const lowerFname = dlFname.toLowerCase();
-        const bz2 =
-          lowerFname.endsWith(".tar.bz2") || lowerFname.endsWith(".tbz");
-        console.log(
-          "Start to extract the tarball archive",
-          dlPath,
-          "to",
-          extractPath
-        );
-        try {
-          await spawnPromise(
-            "tar",
-            [`xvf${bz2 ? "j" : "z"}`, dlPath, "-C", extractPath],
-            { capture: ["stdout", "stderr"] }
-          );
-        } catch (e) {
-          console.error(`${e}:\n`, e.stderr);
-          throw e;
-        }
-        console.log(
-          "The tarball archive",
-          dlPath,
-          "has extracted to ",
-          extractPath
-        );
-      } else {
-        console.warn("Not supported platform.");
-        return;
-      }
+  if (win == null) return;
```

Comment:
Can you add some logs here?
```suggestion
  if (win == null) {
    console.log("Stop update process because win is null.");
    return;
  }
```